test/unit/lib/runner/writers/action-json-writer-tests.js

```diff
@@ -0,0 +1,798 @@
+      actionJsonWriter._getActionTestData.returns({
+        count: 0,
+      });
+      configHandler.get.returns('./testPath');
+
+      actionJsonWriter.write(report);
+
+      expect(path.resolve.args).to.deep.equal([['./testPath']]);
+    });
+
+    it('should call fs.readFileSync with the path to the test and \'utf8\'', function() {
+      path.resolve.returns('./testpath');
+      actionJsonWriter._getActionTestData.returns({
+        count: 0,
+      });
+
+      actionJsonWriter.write(report);
+
+      expect(fs.readFileSync.args).to.deep.equal([
+        [
+          './testpath/testOne',
+          'utf8',
+        ],
+      ]);
+    });
+
+    it('should call actionJsonWriter._getActionTestData with action name passed in', function() {
+      actionJsonWriter._getActionTestData.returns({
+        count: 0,
+      });
+
+      actionJsonWriter.write(report);
+
+      expect(actionJsonWriter._getActionTestData.args).to.deep.equal([['navigate.navigate'], ['article1.click']]);
+    });
+
+    it('should call actionJsonWriter._checkActionForError with action and testData passed in', function() {
+      actionJsonWriter._getActionTestData.returns({
+        count: 0,
+      });
+
+      actionJsonWriter.write(report);
+
+      expect(actionJsonWriter._checkActionForError.args).to.deep.equal([
+        [
+          {
+            'action': 'navigate',
+            'component': 'navigate',
+            'status': 'pass',
+            'steps': {
+              'effects': {
+                'error': null,
+                'status': 'pass',
+              },
+              'perform': {
+                'error': null,
+                'status': 'pass',
+              },
+              'precondition': {
+                'error': null,
+                'status': 'pass',
+              },
+            },
+            'time': [
+              1,
+              201697760,
+            ],
+          },
+          {
+            'averageTime': NaN,
+            'count': 2,
+          },
+        ],
+        [
+          {
+            'action': 'click',
+            'component': 'article1',
+            'status': 'fail',
+            'steps': {
+              'perform': {
+                'error': 'perform failed',
+                'status': 'fail',
+              },
+            },
+            'time': [
+              1,
+              201697760,
+            ],
+          },
+          {
+            'averageTime': NaN,
+            'count': 2,
+          },
+        ],
+      ]);
+    });
+
+    describe('if there is an error in test data', function() {
+      it('should push the error to to testData.failures', function() {
+        actionJsonWriter._getActionTestData.returns({
+          count: 0,
+          failures: [],
+        });
+
+        actionJsonWriter._checkActionForError.returns({error: 'error'});
+
+        actionJsonWriter.write(report);
+
+        expect(actionJsonWriter._checkActionForError.args).to.deep.equal([
```

Comment:
this is checking the arguments for _checkActionForError but your test says its `should push the error to to testData.failures` which is not what this test is testing

Reply:
You will need a test to check that testData.count was incremented, as well the change to average time. In addition you need to check the calls for checkActionForError, and that again the thing was pushed on